### ISSUE TITLE
reportUnknownArgumentType when argument type is defined in Callable via Protocol

### ISSUE THREAD

**Issue by Kein** [microsoft/pylance-release#754]
## Environment data
Pylance 2020.12.2
Python 3.8.5

## Expected behaviour
No warnings

## Actual behaviour
False-positive warning:
![image](https://user-images.githubusercontent.com/623756/102651024-d7f77100-417c-11eb-8dc5-e8f4286d5a3e.png)

## Code Snippet / Additional information

```python
from dataclasses import dataclass
from typing import NamedTuple, Dict, Tuple, Union, Any, Protocol

VertData = NamedTuple("VertData", [('val', int)])
Prop = NamedTuple("Prop", [('size', int)])

class Reader(Protocol):
    def __call__(self, x: memoryview, y: Dict[str, Any]) -> Union[int, Tuple[float,...], VertData, None]:
        ...

@dataclass
class ParsedData:
    prop: Prop
    value: Union[int, Tuple[float,...], VertData, None]
    reader: Reader

data = memoryview(b'sadfsdfsdfsdf')
propMap = Prop(1)
propValues=dict(str1=ParsedData(prop=propMap, value=None, reader=lambda x,y: print(x)))
 
for (k,v) in propValues.items():
    vprp = v.prop
    f = v.reader
    v.value = f(data[1:0], propValues)
```

https://mypy-play.net/?mypy=latest&python=3.7&flags=strict&gist=202f4a912fb9b7125722676630894b6d

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.1.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202110-6-january-2021
